### PR TITLE
Improve MorphTrack API

### PR DESCRIPTION
**Pull request by danielperano**
The MorphTrack's current API has some inconsistencies with TransformTrack's API and some limitations as well (no ability to get/set the desired number of morph targets). This patch makes MorphTrack's API consistent with TransformTrack's API and adds a getter/setter for the number of morph targets.

### CODE REVIEW

**Comment by stephengold** [comment]
Thanks for the contribution.

File: jme3-core/src/main/java/com/jme3/anim/MorphTrack.java
Comment (stephengold) [inline jme3-core/src/main/java/com/jme3/anim/MorphTrack.java:137]: The "alias create" comment applies here, too.

**Review by stephengold — COMMENTED** [review]
Nice work. I noted one javadoc detail that should be restored.

One undocumented/unchecked assumption is that the number of weights equals the number of frames times the number of morph targets. (This becomes clearer if you look at FrameInterpolator.java.) The pre-existing assertion in `setKeyframesWeight()` is wrong! It seems risky to allow changing the number of morph targets without supplying new weights. Could you change `setNbMorphTargets()` to require new weights?

**Comment by danielperano** [comment]
> One undocumented/unchecked assumption is that the number of weights equals the number of frames times the number of morph targets.

This seems to be a critical condition that will cause either erratic behavior or array-out-of-bounds errors if it does not hold. Rather than check this in an assertion, I think we should check that the invariant holds when setting the weights in `setKeyframesWeight()` - setting weights should happen extremely infrequently, so I don't see a performance objection to checking this in an `if` branch rather than an assertion.

File: jme3-core/src/main/java/com/jme3/anim/MorphTrack.java
Comment (danielperano) [reply jme3-core/src/main/java/com/jme3/anim/MorphTrack.java:137]: Noted and applied both here and to the new `weights` parameter of `setNbMorphTargets()`.

**Comment by stephengold** [comment]
Good thinking. I'll review this again once the build error at line 168 is resolved.

**Comment by danielperano** [comment]
Lovely, not sure how I missed that - IDEA decided to stop doing checks, apparently. It's fixed now.

**Comment by stephengold** [comment]
Won't `setNbMorphTargets()` usually cause the assert to fail now?

Also, I thought you were going to change the `assert` to an `if`. Did you change your mind?

**Comment by danielperano** [comment]
I was holding off on the checks changes until hearing back from you since it's a departure from current convention. It's done now.

File: jme3-core/src/main/java/com/jme3/anim/MorphTrack.java
Comment (stephengold) [inline jme3-core/src/main/java/com/jme3/anim/MorphTrack.java:105]: The exception type can be more specific. I advise using `IllegalArgumentException`. I encourage you to also improve line 102 along those lines.

File: jme3-core/src/main/java/com/jme3/anim/MorphTrack.java
Comment (stephengold) [inline jme3-core/src/main/java/com/jme3/anim/MorphTrack.java]: This assertion is now 100% redundant and should be deleted.

File: jme3-core/src/main/java/com/jme3/anim/MorphTrack.java
Comment (stephengold) [inline jme3-core/src/main/java/com/jme3/anim/MorphTrack.java:170]: should also be an `IllegalArgumentException`

**Review by stephengold — CHANGES_REQUESTED** [review]
Better, but still not ready for integration in my opinion.

File: jme3-core/src/main/java/com/jme3/anim/MorphTrack.java
Comment (danielperano) [reply jme3-core/src/main/java/com/jme3/anim/MorphTrack.java:105]: Agreed - I wrote this out with `IllegalArgumentException` at first but changed to the current `RuntimeException` since that's what's currently used. `TransformTrack` also similarly misuses `RuntimeException`, but since the change will have little to no effect on existing exception handlers I think we should change all similar uses to `IllegalArgumentException`.

File: jme3-core/src/main/java/com/jme3/anim/MorphTrack.java
Comment (stephengold) [inline jme3-core/src/main/java/com/jme3/anim/MorphTrack.java:160]: Please delete the word "desired".

File: jme3-core/src/main/java/com/jme3/anim/MorphTrack.java
Comment (stephengold) [inline jme3-core/src/main/java/com/jme3/anim/MorphTrack.java:92]: Please change "weight" to "weights".

**Review by stephengold — CHANGES_REQUESTED** [review]

**Comment by stephengold** [comment]
Thanks again, Daniel.
I think this PR is ready to be integrated, but I'll leave it open for 24 hours in case someone else wants to review it.
I believe this will resolve one of our old issues: #1089.

**Review by stephengold — APPROVED** [review]

**Comment by Ali-RS** [comment]
It Will be nice to fix the formatting issues before merging. (the missing spaces in `if(` and `){`)

**Comment by stephengold** [comment]
You're right, @Ali-RS.

Don't sweat this one, @danielperano --- I'll standardize the formatting for you.

**Comment by stephengold** [comment]
@Ali-RS please verify that the changed lines are now formatted per our standards

**Comment by Ali-RS** [comment]
Looks good to me. Thanks @stephengold

**Pull request merged**
